codersdk/deployment_test.go

```diff
@@ -568,3 +568,115 @@ func TestPremiumSuperSet(t *testing.T) {
 	require.NotContains(t, enterprise.Features(), "", "enterprise should not contain empty string")
 	require.NotContains(t, premium.Features(), "", "premium should not contain empty string")
 }
+
+func TestEmailValuesTakeCorrectPrecedent(t *testing.T) {
```

Comment:
Non-blocking nit: this feels out of place in this PR; perhaps we should land your previous PR (https://github.com/coder/coder/pull/15482) with just the tests rather than muddying this PR with unrelated changes.

Reply:
Sure thing, I'll back these changes out from this PR.